File: javaScript/dtw.js
function dynamicTimeWarping (s1, s2) {
    var n = s1.length + 1;
    var m = s2.length + 1;
    var d = new Array(n);
    for (var i = 0; i < n; i++) {
        d[i] = new Array(m);
        for (var j = 0; j < m; j++) {
            d[i][j] = 0;
        }
    }
    for (var i = 0; i < n; i++) {
        d[i][0] = Infinity;
    }
    for (var j = 0; j < m; j++) {
        d[0][j] = Infinity;
    }
    
    d[0][0] = 0;

    for (var i = 1; i < n; i++) {
        for (var j = 1; j < m; j++) {
            d[i][j] = Math.abs(s1[i-1] - s2[j-1]) 
            + Math.min(d[i-1][j], d[i][j-1], d[i-1][j-1]);
        }
    }
    return d[n-1][m-1];
}

a = [0,2,0,1,0,0]

b = [0,0,0.5,2,0,1,0]

// $.get("textFile.txt", function(data) {
//     var items = data.split("\r\n").map(function(el){ return el.split(",");});
// });

testValue = dynamicTimeWarping(a,b)

console.log(testValue)

File: go/dtw.go
package main

import (
	"fmt"
	"math"
)

func dynamicTimeWarping(s1, s2 []float64) float64 {
	n := len(s1) + 1
	m := len(s2) + 1
	d := make([][]float64, n)
	for i := 0; i < n; i++ {
		d[i] = make([]float64, m)
	}
	for i := 0; i < n; i++ {
		d[i][0] = math.Inf(1)
	}

	for i := 0; i < m; i++ {
		d[0][i] = math.Inf(1)
	}

	d[0][0] = 0

	for i := 1; i < n; i++ {
		for j := 1; j < m; j++ {
			d[i][j] = math.Abs(s1[i-1]-s2[j-1]) +
				math.Min(d[i-1][j-1], math.Min(d[i-1][j], d[i][j-1]))
		}
	}
	return d[n-1][m-1]
}

func main() {
	b := []float64{0.0, 0.0, 0.5, 2.0, 0.0, 1, 0.0}
	a := []float64{0.0, 2.0, 0.0, 1.0, 0.0, 0.0}
	test := dynamicTimeWarping(a, b)
	fmt.Printf("%f", test)
}
